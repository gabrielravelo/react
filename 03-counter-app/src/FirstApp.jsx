import PropTypes from 'prop-types';

const newMessage = {
    title: 'Hola Mundo',
};

const getSaludo = () => {
    return 'Hola Mundo';
}

export const FirstApp = ({ title, subTitle }) => {
  
    return (
        <>
            <h1>{ title }</h1>
            <p>{ subTitle }</p>
            {/* <h1>{ JSON.stringify( newMessage ) }</h1> */}
            <p>Soy un subtitulo</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string
}

