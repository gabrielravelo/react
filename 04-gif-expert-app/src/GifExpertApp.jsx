import { useState } from "react"
import { AddCategory } from "./components/AddCategory";


export const GifExpertApp = () => {

  const [ categories, setCategories ] = useState([ 'One Piece', '2do elemento' ]);

  const onAddCategory = () => {
    setCategories([ 'Valorant', ...categories ]);
    // setCategories( cat => [ ...cat, 'Valorant' ] );
  };

  // console.log( categories );

  return (
    <>
        {/* titulo */}
        <h1>GifExpertApp</h1>

        {/* input */}
        <AddCategory />
        {/* <input /> */}

        {/* Listado de Gif */}
        <button onClick={ onAddCategory }>
          Agregar
        </button>
        <ol>
          { 
            categories.map( category => {
              return <li key={ category } >{ category }</li>
            }) 
          }
        </ol>
            {/* Gif item */}

        
    </>
  )
}

