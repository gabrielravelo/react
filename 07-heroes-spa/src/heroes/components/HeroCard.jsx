import { Link } from 'react-router';

const CharactersByHero = ({alter_ego, characters}) => {
    if ( alter_ego === characters ) return (<></>);
    return (<p>{ characters }</p>);
}

export const HeroCard = ({
    id,
    superhero,
    publisher,
    alter_ego,
    first_appearance,
    characters,
}) => {
    
    const heroImageUrl =  `https://gabrielravelo.github.io/heroes-react-spa/heroes/${ id }.jpg`;
    // const charactersByHero = (<p>{ characters }</p>);

    return (
        <div className="col animate__animated animate__fadeInUp">
            <div className="card">
                
                <div className="row no-gutter">
                    <div className="col-4">
                        <img src={ heroImageUrl } className="card-img" alt={ superhero } />
                    </div>

                    <div className="col-8">
                        <div className="card-body">
                            <h5 className="card-title">{ superhero }</h5>

                            <p className="card-text">{ alter_ego }</p>
                            
                            {/* {
                                ( alter_ego !== characters ) && charactersByHero
                                ( alter_ego !== characters ) && (<p>{ characters }</p>)
                            } */}

                            <CharactersByHero alter_ego={ alter_ego } characters={ characters } />

                            <p className="card-text">
                                <small className="text-muted">{ first_appearance }</small>
                            </p>

                            <Link to={`/hero/${ id }`}>
                                MÃ¡s...
                            </Link>
                        </div>
                    </div>
                </div>
            
            </div>
        </div>
    )
}
