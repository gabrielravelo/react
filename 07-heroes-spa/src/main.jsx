import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import './styles.css'
import { HeroesApp } from './HeroesApp';
import { BrowserRouter } from 'react-router';
import 'animate.css';

createRoot(document.getElementById('root')).render(
  <StrictMode>
    <BrowserRouter>
      <HeroesApp />
    </BrowserRouter>
  </StrictMode>,
)
