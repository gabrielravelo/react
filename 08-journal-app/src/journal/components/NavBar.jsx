import { LogoutOutlined, MenuOutlined } from '@mui/icons-material';
import { AppBar, Grid2, IconButton, Toolbar, Typography } from '@mui/material';

export const NavBar = ({ drawerWith = 240 }) => {
    return (
        <AppBar 
            position='fixed'
            sx={{ 
                width: { sm: `calc(100% - ${ drawerWith }px)` },
                ml: { sm: `${ drawerWith }px` }
             }}
        >
            <Toolbar>
                <IconButton
                    color='inherit'
                    edge='start'
                    sx={{ mr: 2, display: { sm: 'none'} }}
                >
                    <MenuOutlined />
                </IconButton>

                <Grid2 sx={{ width: '100%' }} container direction='row' justifyContent='space-between' alignItems='center'>
                    <Typography variant='h6' noWrap component='div'>JournalApp</Typography>

                    <IconButton color='error'>
                        <LogoutOutlined />
                    </IconButton>
                </Grid2>
            </Toolbar>
        </AppBar>
    )
}
