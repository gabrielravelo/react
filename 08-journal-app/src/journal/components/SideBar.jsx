import { TurnedInNot } from "@mui/icons-material"
import { Box, Divider, Drawer, Grid2, List, ListItem, ListItemButton, ListItemIcon, ListItemText, Toolbar, Typography } from "@mui/material"
import { useSelector } from "react-redux"

export const SideBar = ({ drawerWith}) => {

    const { displayName = 'Name Not Found' } = useSelector( state => state.auth );

    return (
        <Box
            component='nav'
            sx={{ width: { sm: drawerWith }, flexShrink: { sm: 0 } }}
        >
            <Drawer
                variant="permanent"
                open
                sx={{ 
                    display: { xs: 'block' },
                    '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWith } 
                }} 
            >
                <Toolbar>
                    <Typography variant="h6" noWrap component='div'>
                        { displayName }
                    </Typography>
                </Toolbar>
                <Divider />

                <List>
                    {
                        ['Enero', 'Febrero', 'Marzo', 'Abril'].map( text => (
                            <ListItem key={ text } disablePadding>
                                <ListItemButton>
                                    <ListItemIcon>
                                        <TurnedInNot />
                                    </ListItemIcon>

                                    <Grid2 container>
                                        <ListItemText primary={ text } />
                                        <ListItemText secondary='loremadmwajdad jdawdjawdj ojadiowa jjdjaw ajdwaida' />
                                    </Grid2>
                                </ListItemButton>
                            </ListItem>
                        ))
                    }
                </List>

            </Drawer>

        </Box>
    )
}
