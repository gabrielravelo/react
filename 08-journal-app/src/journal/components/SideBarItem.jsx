import { TurnedInNot } from '@mui/icons-material'
import { Grid2, ListItem, ListItemButton, ListItemIcon, ListItemText } from '@mui/material';
import { useMemo } from 'react';
import { useDispatch } from 'react-redux';
import { setActiveNote } from '../../store/journal';


export const SideBarItem = ({ title = '', body, id, date, imageUrls = [] }) => {
    
    const dispatch = useDispatch();

    const newTitle = useMemo( () => {
        return title.length > 15
            ? title.substring(0,15) + '...'
            : title;
    }, [ title ]);

    const onClickNote = () => {
        dispatch( setActiveNote({title, body, id, date, imageUrls}) )
    }

    return (
        <ListItem
            onClick={ onClickNote }
            disablePadding
        >
            <ListItemButton>
                <ListItemIcon>
                    <TurnedInNot />
                </ListItemIcon>

                <Grid2 container>
                    <ListItemText primary={ newTitle } />
                    <ListItemText secondary={ body } />
                </Grid2>
            </ListItemButton>
        </ListItem>
    )
}
