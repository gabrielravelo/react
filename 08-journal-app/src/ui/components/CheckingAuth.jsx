
import { CircularProgress, Grid2 } from '@mui/material';

export const CheckingAuth = () => {
    return (
        <Grid2
            container
            spacing={ 0 }
            direction="column"
            alignItems="center"
            justifyContent="center"
            sx={{ minHeight: '100vh', backgroundColor: 'primary.main', padding: 4 }}
        >
            <Grid2
                // size={ 8 }
                sx={{ 
                    // backgroundColor: 'white', 
                    // padding: 3, 
                    // borderRadius: 2, 
                    width: { md: 450 }  
                }} 
            >
                <CircularProgress color='warning' />
            </Grid2>
        </Grid2>
    )
}
