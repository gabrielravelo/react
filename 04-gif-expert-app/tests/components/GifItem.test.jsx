import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';


describe('Pruebas en <GifItem />', () => {

    const title = 'Gabo';
    const url = 'https://gabrielravelo.com/';

    test('Debe de hacer match con el snapshot', () => {
        const { container } = render( <GifItem title={ title } url={ url } /> );
        expect( container ).toMatchSnapshot();

    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render( <GifItem title={ title } url={ url } /> );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url);
        expect( alt ).toBe( title );
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );
    });

    test('debe de mostrar el titulo en el componente', () => {
        render( <GifItem title={ title } url={ url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();
    });

});