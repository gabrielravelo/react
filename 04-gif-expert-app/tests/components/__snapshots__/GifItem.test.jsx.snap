// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gabo"
      src="https://gabrielravelo.com/"
    />
    <p>
      Gabo
    </p>
  </div>
</div>
`;
